crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -1014,89 +1023,130 @@ struct RangeBlockMode<
+    pending_tasks: FuturesOrdered<ReceiptFetchFuture<Eth::Provider>>,
 }
 
 impl<
         Eth: RpcNodeCoreExt<Provider: BlockIdReader, Pool: TransactionPool> + EthApiTypes + 'static,
     > RangeBlockMode<Eth>
 {
     async fn next(&mut self) -> Result<Option<ReceiptBlockResult<Eth::Provider>>, EthFilterError> {
+        // First, try to return any already processed results
         if let Some(result) = self.next.pop_front() {
             return Ok(Some(result));
         }
 
-        let Some(next_header) = self.iter.next() else {
-            return Ok(None);
-        };
+        // Next, try to get a completed task result if there are pending tasks
+        if !self.pending_tasks.is_empty() {
+            if let Some(task_result) = self.pending_tasks.next().await {
+                match task_result {
+                    Ok(chunk_results) => {
+                        for result in chunk_results {
+                            self.next.push_back(result);
+                        }
+                        return Ok(self.next.pop_front());
+                    }
+                    Err(e) => return Err(e),
+                }
+            }
+        }
+
+        // If no pending tasks and no queued results, try to start new tasks
+        if self.pending_tasks.is_empty() {
+            // Try to get the next header(s) to process
+            let Some(next_header) = self.iter.next() else {
+                return Ok(None);
+            };
+
+            let mut range_headers = Vec::with_capacity(self.max_range);
+            range_headers.push(next_header);
+
+            // Collect consecutive blocks up to max_range size
+            while range_headers.len() < self.max_range {
+                let Some(peeked) = self.iter.peek() else { break };
+                let Some(last_header) = range_headers.last() else { break };
 
-        let mut range_headers = Vec::with_capacity(self.max_range);
-        range_headers.push(next_header);
+                let expected_next = last_header.number() + 1;
+                if peeked.number() != expected_next {
+                    break; // Non-consecutive block, stop here
+                }
 
-        // Collect consecutive blocks up to max_range size
-        while range_headers.len() < self.max_range {
-            let Some(peeked) = self.iter.peek() else { break };
-            let Some(last_header) = range_headers.last() else { break };
+                let Some(next_header) = self.iter.next() else { break };
+                range_headers.push(next_header);
+            }
 
-            let expected_next = last_header.header().number() + 1;
-            if peeked.header().number() != expected_next {
-                break; // Non-consecutive block, stop here
+            // Check if we should use parallel processing for large ranges
+            let remaining_headers = self.iter.len() + range_headers.len();
+            if remaining_headers >= PARALLEL_PROCESSING_THRESHOLD as usize {
```

Comment:
wondering if `PARALLEL_PROCESSING_THRESHOLD` can be a `usize` const so this cast isn't needed